### PR TITLE
implements quickest approach to supporting permission levels on stude…

### PR DESCRIPTION
**Pull request by esotrope**
…nt fields in advanced filters

### CODE REVIEW

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettings.java [hunk 1]
```diff
@@ -223,7 +223,7 @@ public void setStudentFields(final Map<StudentFieldType, SearchFieldPermissionLe
      * @deprecated use {@link #getStudentFields()}}
      */
     public boolean isLimitedEnglishProficienciesEnabled() {
-        return isStudentFieldEnabled(StudentFieldType.LimitedEnglishProficiencies);
+        return isStudentFieldEnabled(StudentFieldType.LimitedEnglishProficiency);
```
Comment (esotrope) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettings.java]: unrelated naming fix

File: webapp/src/main/webapp/src/app/assessments/filters/adv-filters/adv-filters.component.html [hunk 2]
```diff
@@ -113,140 +160,228 @@
     <!-- Column 1 -->
     <div class="col-md-3">
       <!-- Gender -->
-      <div class="block-children">
+      <div class="block-children" *ngIf="isEnabled('Gender')">
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/filters/adv-filters/adv-filters.component.html]: only real change was to add these if-checks on all the student fields

File: webapp/src/main/webapp/src/app/assessments/filters/adv-filters/adv-filters.component.ts [hunk 3]
```diff
@@ -32,18 +38,14 @@ export class AdvFiltersComponent {
   private sbTypeAheadGroup: SBTypeaheadGroup;
 
   showTransferAccess = false;
-  showElas = false;
-  showLep = false;
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/filters/adv-filters/adv-filters.component.ts]: no longer needed as this is inferred from the available student fields

**Comment by esotrope** [comment]
image

what teachers will see when the settings are in place

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/model/StudentFieldType.java [hunk 4]
```diff
@@ -1,15 +1,15 @@
 package org.opentestsystem.rdw.reporting.common.model;
 
 /**
- * Declares all student filter types.
+ * Declares all student field types.
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/model/StudentFieldType.java]: doh, thanks
